atc/job_config.go

```diff
@@ -10,6 +10,7 @@ type JobConfig struct {
 	SerialGroups         []string `yaml:"serial_groups,omitempty" json:"serial_groups,omitempty" mapstructure:"serial_groups"`
 	RawMaxInFlight       int      `yaml:"max_in_flight,omitempty" json:"max_in_flight,omitempty" mapstructure:"max_in_flight"`
 	BuildLogsToRetain    int      `yaml:"build_logs_to_retain,omitempty" json:"build_logs_to_retain,omitempty" mapstructure:"build_logs_to_retain"`
+	BuildLogsDaysToRetain int      `yaml:"build_logs_days_to_retain,omitempty" json:"build_logs_days_to_retain,omitempty" mapstructure:"build_logs_days_to_retain"`
```

Comment:
What do you think of something like this?:

```yaml
build_log_retention:
  builds: 10
  days: 5
```

We'd have to keep `build_logs_to_retain` for backwards-compatibility, but this would make it easier to add more retention policies without polluting the toplevel configuration too much. I could see someone wanting something like this for example:

```yaml
build_log_retention:
  failed_builds: 1000
  succeeded_builds: 10
```

Whaddaya think?